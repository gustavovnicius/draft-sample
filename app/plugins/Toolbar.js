import React from 'react'
import createInlineToolbarPlugin from 'draft-js-inline-toolbar-plugin'
import ConnectedAddCommentButton from 'containers/ConnectedAddCommentButton'
import 'draft-js-inline-toolbar-plugin/lib/plugin.css'

const Toolbar = createInlineToolbarPlugin({
  structure: [
    ConnectedAddCommentButton,
  ]
})

export default Toolbar
