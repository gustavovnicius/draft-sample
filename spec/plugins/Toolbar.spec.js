import React from 'react'
import renderer from 'react-test-renderer'
jest.mock('containers/ConnectedAddCommentButton')

import Toolbar from 'plugins/toolbar';

describe('Toolbar', () => {
  describe('plugin object', () => {
    it('matches the expected structure', () => {
      const expected = {
        InlineToolbar: expect.any(Function),
        initialize: expect.any(Function),
        onChange: expect.any(Function)
      }

      expect(Toolbar).toEqual(expected)
    })
  })

  describe('decorated component', () => {
    it('matches the snapshot', () => {
      const component = renderer.create(
        <Toolbar.InlineToolbar />
      )
      const tree = component.toJSON();

      expect(tree).toMatchSnapshot();
    })
  })
})
