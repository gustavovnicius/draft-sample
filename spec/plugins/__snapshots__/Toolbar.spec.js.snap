exports[`Toolbar decorated component matches the snapshot 1`] = `
<div
  className="draftJsToolbar__toolbar__dNtBH"
  style={undefined} />
`;
